data-prepper-api/src/main/java/org/opensearch/dataprepper/model/acknowledgements/AcknowledgementSetManager.java

```diff
@@ -30,6 +30,7 @@ public interface AcknowledgementSetManager {
      */
     AcknowledgementSet create(final Consumer<Boolean> callback, final Duration timeout);
 
+    AcknowledgementSet create(final Consumer<Double> progressCheckCallback, final Duration progressCheckInterval, final Consumer<Boolean> callback, final Duration timeout);
```

Comment:
Since the progress check is optional, it might make sense to add the progress check on the `AcknowledgementSet` itself. This will help with clarity in the caller's code.

e.g.

```
acknowledgementSet = manager.create(myCallback, timeout);
acknowledgementSet.addProgressCheck(myProgressCallback, progressInterval);
```

We may also want an expiry callback right? So continuing to add to the `create` method is making the code harder to read. I'm open to other ideas as well - perhaps a builder?

```
manager.builder(myCallback, timeout)
  .withProgressCheck(myProgressCallback, progressInterval)
  .build()
```

Reply:
What if I want an expiry callback, but not progress? This becomes confusing to use. It is also already difficult to read and understand in the SQS code because you need to know the order of these callbacks and durations.